sos/plugins/openstack_novajoin.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2018 Red Hat, Inc., David Vallee Delisle <dvd@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin
+import os
+
+
+class OpenStackNovajoin(Plugin):
+    """OpenStack Novajoin
+    """
+    plugin_name = "openstack_novajoin"
+    profiles = ('openstack', 'openstack_undercloud')
+
+    def setup(self):
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/novajoin/"
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/novajoin/*.log"
+            ])
+
+    def postproc(self):
+        regexp = (r"(?i)(?P<var>(UNDERCLOUD|OS)_[^\s|^=]*[_]*(PASSWORD|SECRET)"
+                  "(_KEY)*=)(.*)")
+        self.do_file_sub("/var/log/novajoin/*.log",
+                         regexp, r"\g<var>*********")
+        self.do_file_sub("/etc/novajoin/novajoin.conf", regexp,
```

Comment:
With large log files (e.g. `--all-logs --log-size 0`) this may become expensive. Does this component really write passwords and other secrets to its logs?

Reply:
I removed the search in the logs because I didn't see anything that looks like a password, but there's one in novajoin.conf. I don't think the log folder will ever grow much. It's not a very verbose service from what I can see.